model/BUILD

```diff
@@ -14,18 +14,41 @@ go_library(
         "//model/proxy/alphav1/config:go_default_library",
         "@com_github_ghodss_yaml//:go_default_library",
         "@com_github_golang_glog//:go_default_library",
+        "@com_github_golang_mock//gomock:go_default_library",
         "@com_github_golang_protobuf//jsonpb:go_default_library",
         "@com_github_golang_protobuf//proto:go_default_library",
         "@com_github_hashicorp_go_multierror//:go_default_library",
     ],
 )
 
+# Auto-generate mock interfaces for unit testing model package.
+#
+# This should be kept in sync with the `go:generate` directive at the
+# top of config_test.go which is necessary for non-bazel testing with
+# `go test`.
+#
+# Add `_test` suffix to the output file so mock interfaces, functions,
+# etc. are excluded from test coverage, gometalinter, etc.
+genrule(
+    name = "genmocks",
+    srcs = ["config.go"],
+    outs = ["mock_config_test.go"],
+    cmd = " mockgen -source $< -destination $@ -package model",
+)
+
 go_test(
     name = "go_default_test",
     size = "small",
     srcs = [
         "config_test.go",
         "service_test.go",
+        "mock_config_test.go",
     ],
+    data = [":genmocks"],
```

Comment:
I think bazel is smart enough to run this rule without specifying it as data dependency.